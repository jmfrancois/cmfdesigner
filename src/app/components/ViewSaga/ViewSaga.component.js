import React from 'react';
import PropTypes from 'prop-types';
import { Inject, cmfConnect } from '@talend/react-cmf';

function ViewSaga(props) {
	return (
		<div>
			<h1>Saga: {props.item.name}</h1>
			<div>
				<ul>
					{(props.item.params.length > 1) && (
						<p>This saga has multiple arguments</p>
					)}
				</ul>
			</div>
			<Inject component="FileAnalytics" path={props.item.path} />
		</div>
	);
}
ViewSaga.displayName = 'ViewSaga';
ViewSaga.propTypes = {
	item: PropTypes.shape({
		name: PropTypes.string,
		params: PropTypes.array,
		path: PropTypes.string,
	}),
};
ViewSaga.defaultProps = {
	item: { params: [] },
};
export default cmfConnect({})(ViewSaga);
