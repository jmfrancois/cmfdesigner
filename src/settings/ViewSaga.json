{
  "props": {
    "ViewSaga#default": {
      "itemExpression": "service#designer.sagas:getCurrent",
      "renderIfExpression": {
        "id": "cmf.router.matchPath",
        "args": [
            {
                "path": "/sagas/view",
                "isExact": true
            }
        ]
      }
    }
  }
}
