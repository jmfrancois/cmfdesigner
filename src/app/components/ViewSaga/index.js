import ViewSaga from './ViewSaga.component';

export default ViewSaga;
