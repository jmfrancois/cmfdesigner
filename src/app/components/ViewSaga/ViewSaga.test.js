import React from 'react';
import { shallow } from 'enzyme';

import ViewSaga from './ViewSaga.component';

describe('ViewSaga', () => {
	it('should render', () => {
		const wrapper = shallow(
			<ViewSaga.WrappedComponent />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
