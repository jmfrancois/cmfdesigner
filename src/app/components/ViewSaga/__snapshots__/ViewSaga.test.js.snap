// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewSaga should render 1`] = `
<div>
  <h1>
    Saga: 
  </h1>
  <div>
    <ul />
  </div>
  <Inject
    component="FileAnalytics"
    path={undefined}
  />
</div>
`;
